### PR TITLE
Prefer Guice constructor injection

### PR DESCRIPTION
**Pull request by elharo**

### CODE REVIEW

File: src/main/java/org/apache/maven/plugins/gpg/SignDeployedMojo.java [hunk 1]
```diff
@@ -106,11 +105,15 @@ public class SignDeployedMojo extends AbstractGpgMojo {
     @Parameter(property = "artifacts")
     private String artifacts;
 
-    @Component
-    private RepositorySystem repositorySystem;
+    private final RepositorySystem repositorySystem;
+
+    private final Map<String, ArtifactCollectorSPI> artifactCollectors;
```
Comment (slachiewicz) [inline src/main/java/org/apache/maven/plugins/gpg/SignDeployedMojo.java:110]: why we need it?

Comment (elharo) [reply src/main/java/org/apache/maven/plugins/gpg/SignDeployedMojo.java:110]: I'm not 100% sure. It is used by the existing code.

Comment (slachiewicz) [reply src/main/java/org/apache/maven/plugins/gpg/SignDeployedMojo.java:110]: but `artifactCollectors` looks like new variable added by You

Comment (elharo) [reply src/main/java/org/apache/maven/plugins/gpg/SignDeployedMojo.java:110]: No. it was just moved a little bit. It's on line 113 in the original.

**Review by slachiewicz — APPROVED** [review]

**Pull request merged**